### DISCUSSION TITLE
Send H264 packets encoded with NvEnc

### DISCUSSION THREAD

**Discussion by vtchek** [paullouisageneau/libdatachannel#928] [Q&A]
Hello, the streamer example seems to fit my needs. I have coded a 3D Engine and I would like to be able to stream the frames rendered to a browser. I have already managed to encode the frames using NvEnc and I am able to write them to an .h264 file and read them trough VLC. I would like to use those packets instead of a directory containing each frame. Is there any example of this? Looking at the code it seems that I only have to tweak the parser files and the loadNextSample to use the packet encoded by NvEnc ( vector<uint8_t> fileContents((std::istreambuf_iterator<char>(source)), std::istreambuf_iterator<char>());) am I wrong? Thanks for your help !

**Comment by jiox** [comment]
Yes, but NVENC ouputs NAL units separated by start codes, not length prefix codes as in the example provided in the repository. So just make sure to initialize the `rtc::H264RtpPacketizer` with `rtc::H264RtpPacketizer::Separator::StartSequence` and you are good to go.

**Comment by vtchek** [comment]
I modified my packets to start by NAL unit length and not the start sequence, but its a bit overkill since what you mentioned exists, thank you !
